mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/bridge/builtin/impl/DefaultJavaUtilDateValueBridge.java

```diff
@@ -27,9 +27,10 @@ public Instant toIndexedValue(Date value, ValueBridgeToIndexedValueContext conte
 			return null;
 		}
 
-		// java.sql.Date does not support toInstant(). See HSEARCH-3670.
-		// We're using here the same strategy applied on DefaultJavaSqlDateValueBridge to handle them.
-		return value instanceof java.sql.Date ? Instant.ofEpochMilli( value.getTime() ) : value.toInstant();
+		// java.sql.Date and java.sql.Time do not support toInstant(). See HSEARCH-3670.
+		// We're using here the same strategy applied on DefaultJavaSql*ValueBridge to handle them.
+		return ( value instanceof java.sql.Date ) | ( value instanceof java.sql.Time ) ?
```

Comment:
You used a bitwise `OR` here (single pipe). Please avoid that.

Also, same comment as before: you could just always use `Instant.ofEpochMilli( value.getTime() )`.